e2e/src/test/java/io/aiven/kafka/tieredstorage/e2e/SingleBrokerTest.java

```diff
@@ -0,0 +1,619 @@
+            assertThat(record.offset()).isEqualTo(newStartOffset);
+        }
+    }
+
+    @Test
+    @Order(4)
+    void remoteCleanupDueToRetention() throws Exception {
+        maybeFailFast();
+
+        // Collect all remote segments, as after changing retention, all should be deleted.
+        final var remoteSegmentsBefore = remoteLogMetadataTracker.remoteSegments();
+        final var segmentsToBeDeleted = Stream.concat(
+            remoteSegmentsBefore.get(t0p0).stream(),
+            remoteSegmentsBefore.get(t0p1).stream()
+        ).collect(Collectors.toList());
+
+        LOG.info("Forcing cleanup by setting bytes retention to 1");
+
+        final var alterConfigs = List.of(new AlterConfigOp(
+            new ConfigEntry("retention.bytes", "1"), AlterConfigOp.OpType.SET));
+        adminClient.incrementalAlterConfigs(Map.of(
+            new ConfigResource(ConfigResource.Type.TOPIC, TOPIC_0), alterConfigs
+        )).all().get(5, TimeUnit.SECONDS);
+
+        LOG.info("Starting cleanup validation");
+
+        try (final var consumer = new KafkaConsumer<>(Map.of(
+            "bootstrap.servers", kafka.getBootstrapServers(),
+            "auto.offset.reset", "earliest"
+        ), new ByteArrayDeserializer(), new ByteArrayDeserializer())) {
+
+            // Get earliest offset available locally
+            final long newStartOffset = localLogFiles(TP_0_0).stream()
+                .mapToLong(f -> Long.parseLong(f.getName().replace(".log", "")))
+                .max()
+                .getAsLong();
+            // and wait til expected earliest offset is in place
+            await()
+                .pollInterval(Duration.ofSeconds(5))
+                .atMost(Duration.ofSeconds(30))
+                .until(() -> {
+                    final var beginningOffset = consumer.beginningOffsets(List.of(TP_0_0)).get(TP_0_0);
+                    LOG.info("Beginning offset found {}, expecting {}", beginningOffset, newStartOffset);
+                    return beginningOffset.equals(newStartOffset);
+                });
+
+            final Map<TopicPartition, Long> endOffsets = consumer.endOffsets(List.of(TP_0_0));
+            assertThat(endOffsets.get(TP_0_0)).isEqualTo(RECORDS_TO_PRODUCE + 1);
+
+            // TODO check for EARLIEST_LOCAL_TIMESTAMP when available in client
+
+            consumer.assign(List.of(TP_0_0));
+            consumer.seek(TP_0_0, 0);
+
+            final ConsumerRecord<byte[], byte[]> record = consumer.poll(Duration.ofSeconds(1)).records(TP_0_0).get(0);
+            assertThat(record.offset()).isEqualTo(newStartOffset);
+
+            remoteLogMetadataTracker.waitUntilSegmentsAreDeleted(segmentsToBeDeleted);
+            assertNoTopicDataOnTierStorage(t0p0.topic(), t0p0.topicId());
+        }
+
+        LOG.info("Cleanup validation completed");
+    }
+
+    @Test
+    @Disabled("https://github.com/aiven/kafka/issues/35")
+    @Order(5)
+    void topicDelete() throws Exception {
```

Comment:
topic delete with tiered storage is still flaky. Disabling it in the meantime, we can troubleshoot this after switching to upstream.

Reply:
```
     [2023-08-23 00:32:07,465] ERROR [RemoteLogManager=1 partition=4OkS_EQaSVKfGYZVj2w9bg:topic1-0] Error occurred while copying log segments of partition: 4OkS_EQaSVKfGYZVj2w9bg:topic1-0 (kafka.log.remote.RemoteLogManager$RLMTask)
    java.lang.IllegalStateException: This instance is in invalid state, initialized: false close: false
    	at org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager.ensureInitializedAndNotClosed(TopicBasedRemoteLogMetadataManager.java:501)
    	at org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager.highestOffsetForEpoch(TopicBasedRemoteLogMetadataManager.java:224)
    	at kafka.log.remote.RemoteLogManager.$anonfun$findHighestRemoteOffset$2(RemoteLogManager.scala:612)
    	at kafka.log.remote.RemoteLogManager.$anonfun$findHighestRemoteOffset$2$adapted(RemoteLogManager.scala:609)
    	at scala.Option.foreach(Option.scala:437)
    	at kafka.log.remote.RemoteLogManager.$anonfun$findHighestRemoteOffset$1(RemoteLogManager.scala:609)
    	at kafka.log.remote.RemoteLogManager.$anonfun$findHighestRemoteOffset$1$adapted(RemoteLogManager.scala:608)
    	at scala.Option.foreach(Option.scala:437)
    	at kafka.log.remote.RemoteLogManager.findHighestRemoteOffset(RemoteLogManager.scala:608)
    	at kafka.log.remote.RemoteLogManager$RLMTask.maybeUpdateReadOffset$1(RemoteLogManager.scala:394)
    	at kafka.log.remote.RemoteLogManager$RLMTask.copyLogSegmentsToRemote(RemoteLogManager.scala:399)
    	at kafka.log.remote.RemoteLogManager$RLMTask.run(RemoteLogManager.scala:576)
    	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
    	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
    	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    	at java.base/java.lang.Thread.run(Thread.java:829)
    [2023-08-23 00:32:07,469] ERROR [RemoteLogManager=1 partition=4OkS_EQaSVKfGYZVj2w9bg:topic1-0] Error while cleaning up log segments for partition: 4OkS_EQaSVKfGYZVj2w9bg:topic1-0 (kafka.log.remote.RemoteLogManager$RLMTask)
    java.lang.IllegalStateException: This instance is in invalid state, initialized: false close: false
    	at org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager.ensureInitializedAndNotClosed(TopicBasedRemoteLogMetadataManager.java:501)
    	at org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager.listRemoteLogSegments(TopicBasedRemoteLogMetadataManager.java:240)
    	at kafka.log.remote.RemoteLogManager$RLMTask.handleExpiredRemoteLogSegments(RemoteLogManager.scala:505)
    	at kafka.log.remote.RemoteLogManager$RLMTask.run(RemoteLogManager.scala:580)
    	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
    	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
    	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    	at java.base/java.lang.Thread.run(Thread.java:829)
    [2023-08-23 00:32:07,467] INFO [Broker id=1] Finished LeaderAndIsr request in 303ms correlationId 1 from controller 1 for 1 partitions (state.change.logger)
```